app/logic/celtsLabor.py

```diff
@@ -1,130 +1,130 @@
+import requests
+import json
+from peewee import DoesNotExist
+from collections import defaultdict
+from app import app
+from app.models.user import User
+from app.models.celtsLabor import CeltsLabor
+from app.models.term import Term
+
+def getCeltsLaborFromLsf():
+    """
+        Make a call to the LSF endpoint which returns all CELTS student labor records. 
+
+        The returned data is a dictionary with B# key and value that is a list of dicts that contain the labor information. 
+
+    """
+    try: 
+        lsfUrl = f"{app.config['lsf_url'].strip('/')}/api/org/2084"
+        response = requests.get(lsfUrl)
+        return response.json()
+    except json.decoder.JSONDecodeError: 
+        print(f'Response from {lsfUrl} was not JSON.\n' + response.text)
+        return {}
+    except KeyError as e: 
+        print(f'Make sure you have "lsf_url" set in your local-override config file.')
+        raise(e)
+
+def updateCeltsLaborFromLsf():
+    """
+        Parse the LSF response object so that it is only labor records for summer and academic years.
+
+        Input: JSON response
+                {'B#': [{'positionTitle': 'Position Title',
+                         'termCode': 'Term Code', 
+                         'laborStart': Labor Start Date, 
+                         'laborEnd': Labor End Date, 
+                         'jobType': 'Job Type',
+                         'wls': 'WLS Lvl',
+                         'termName': 'Term Name'}],
+                 'B#': [{'positionTitle': 'Position Title',
+                         'termCode': 'Term Code', 
+                         'laborStart': Labor Start Date, 
+                         'laborEnd': Labor End Date, 
+                         'jobType': 'Job Type',
+                         'wls': 'WLS Lvl',
+                         'termName': 'Term Name'}]}
+    """
+    laborDict = getCeltsLaborFromLsf()
+    
+    studentLaborDict = {}
+    for key, value in laborDict.items(): 
+        try: 
+            username = User.get(bnumber = key)
+            # All term codes for summer end with 13 and all term codes for an academic year end in 00 and those are the only terms we want to record.
+            studentLaborDict[username] = collapsePositions([p for p in value if str(p["termCode"])[-2:] in ["00","13"]])
+        except DoesNotExist: 
+            pass
+
+    refreshCeltsLaborRecords(studentLaborDict)
+
+def collapsePositions(positionList):
+    '''
+        Parse position information from the JSON response and return only the position title and term title 
+
+        Input: [{'positionTitle': 'Position Title',
+                 'termCode': 'Term Code', 
+                 'laborStart': Labor Start Date, 
+                 'laborEnd': Labor End Date, 
+                 'jobType': 'Job Type',
+                 'wls': 'WLS Lvl',
+                 'termName': 'Term Name'}]
+        
+        Returned:{'Position Title': ['Term Name']}
+
+    '''
+    positionDict = defaultdict(list)
```

Comment:
This could maybe be changed into something like laborPositionDict to be more descriptive